### PR TITLE
Update project dependencies

### PR DESCRIPTION
**Pull request by rm-hull**

### CODE REVIEW

**Comment by lvh** [comment]
Hm, that's an interesting failure. It might be worth adding `pjstadig/humane-test-output` to make Travis output nicer; even if it's in a different branch first :)

**Comment by rm-hull** [comment]
So in moving from clojure 1.4 to 1.8 there are clearly some differences in how macros are expanded.

Here is a formatted test output (via https://github.com/venantius/ultra):

```clojure
FAIL in (test-wrap-map) (test_coverage.clj:61)

expected: (expand=
           (clojure.core/seq
            (clojure.core/concat
             (clojure.core/list (quote cloverage.coverage/capture))
             (clojure.core/list 0)
             (clojure.core/list
              (clojure.core/apply
               clojure.core/hash-map
               (clojure.core/seq
                (clojure.core/concat
                 (clojure.core/list
                  (clojure.core/seq
                   (clojure.core/concat
                    (clojure.core/list (quote cloverage.coverage/capture))
                    (clojure.core/list 3)
                    (clojure.core/list :a))))
                 (clojure.core/list
                  (clojure.core/seq
                   (clojure.core/concat
                    (clojure.core/list (quote cloverage.coverage/capture))
                    (clojure.core/list 4)
                    (clojure.core/list (quote apple)))))
                 (clojure.core/list
                  (clojure.core/seq
                   (clojure.core/concat
                    (clojure.core/list (quote cloverage.coverage/capture))
                    (clojure.core/list 1)
                    (clojure.core/list :b))))
                 (clojure.core/list
                  (clojure.core/seq
                   (clojure.core/concat
                    (clojure.core/list (quote cloverage.coverage/capture))
                    (clojure.core/list 2)
                    (clojure.core/list (quote banana)))))))))))
           (wrap (var track-coverage) 0 (quote {:a apple, :b banana})))
  actual: (not
           (expand=
            (cloverage.coverage/capture
             0
             {(cloverage.coverage/capture 1 :b) (cloverage.coverage/capture 2 banana),
              (cloverage.coverage/capture 3 :a) (cloverage.coverage/capture 4 apple)})
            (cloverage.instrument/wrapm
             cloverage.coverage/track-coverage
             0
             {:a apple, :b banana})))
```

**Comment by rm-hull** [comment]
This is the test that fails: 
```clojure
(deftest test-wrap-map
  (is (expand= `(capture 0 {(capture 3 :a) (capture 4 ~'apple)
                            (capture 1 :b)  (capture 2 ~'banana)})
               (wrap #'track-coverage 0 '{:a apple :b banana}))))
```
Looking at the other tests, the index (first arg of `capture`) increment sequentially (e.g. 

```clojure
(deftest test-wrap-vector
  (is (expand= `(capture 0
                         [(capture 1 1)
                          (capture 2 "foo")
                          (capture 3 ~'bar)])
               (wrap #'track-coverage 0 '[1 "foo" bar]))))
```

The `test-wrap-map` does not (which seems a bit odd to me) - the args are slightly back-to-front (`:b` and `banana` are counted before `:a` and `apple`).

The test fails because the two expanded expressions are not quite the same, because of the ordering
```
(do (cloverage.coverage/cover 0) {(do (cloverage.coverage/cover 3) :a) (do (cloverage.coverage/cover 4) apple), (do (cloverage.coverage/cover 1) :b) (do (cloverage.coverage/cover 2) banana)})

(do (cloverage.coverage/cover 0) {(do (cloverage.coverage/cover 1) :a) (do (cloverage.coverage/cover 2) apple), (do (cloverage.coverage/cover 3) :b) (do (cloverage.coverage/cover 4) banana)})
```

Is this a problem with the way older clojure versions handled hash-maps, possibly?

To me, it seems sensible to flip that test:

```patch
diff --git a/cloverage/test/cloverage/test_coverage.clj b/cloverage/test/cloverage/test_coverage.clj
index 499adb0..f94fcdf 100644
--- a/cloverage/test/cloverage/test_coverage.clj
+++ b/cloverage/test/cloverage/test_coverage.clj
@@ -58,8 +58,8 @@
                (wrap #'track-coverage 0 '[1 "foo" bar]))))
 
 (deftest test-wrap-map
-  (is (expand= `(capture 0 {(capture 3 :a) (capture 4 ~'apple)
-                            (capture 1 :b)  (capture 2 ~'banana)})
+  (is (expand= `(capture 0 {(capture 1 :a) (capture 2 ~'apple)
+                            (capture 3 :b)  (capture 4 ~'banana)})
                (wrap #'track-coverage 0 '{:a apple :b banana}))))
 
 (deftest test-wrap-list
```

@JacekLach do you have any historical insight to this?

**Comment by JacekLach** [comment]
I believe the indexes were simply taken from whatever order the elements were actually visited in practice; they're not the meaningful part of the test. Your guess that the iteration order changed in clojure is as good as mine.

The relevant code is in [instrument.clj#236](https://github.com/lshift/cloverage/blob/master/cloverage/src/cloverage/instrument.clj#L236):
```
                      (map? form) (zipmap
                                   (doall (map (wrapper f line) (keys form)))
                                   (doall (map (wrapper f line) (vals form))))
```

It sounds reasonable that the default order of `keys` and `vals` has changed between versions.

The change LGTM.

**Comment by lvh** [comment]
(Thanks a lot for the continued effort and the detailed explanations. This was very easy to follow.)

File: cloverage/test/cloverage/test_coverage.clj [hunk 1]
```diff
@@ -58,8 +58,8 @@
                (wrap #'track-coverage 0 '[1 "foo" bar]))))
 
 (deftest test-wrap-map
-  (is (expand= `(capture 0 {(capture 3 :a) (capture 4 ~'apple)
-                            (capture 1 :b)  (capture 2 ~'banana)})
+  (is (expand= `(capture 0 {(capture 1 :a) (capture 2 ~'apple)
+                            (capture 3 :b)  (capture 4 ~'banana)})
```
Comment (JacekLach) [inline cloverage/test/cloverage/test_coverage.clj]: nit: let's fix the double space in the second line since we're touching this part anyway? :P

Comment (rm-hull) [inline cloverage/test/cloverage/test_coverage.clj]: -squashed-

**Comment by lvh** [comment]
Tests passed in master, so let's see if that still works. I'm building up a changelog in my branch so we can do the SNAPSHOT testing you mentioned.

**Pull request merged**